File: ecommerce-frontend/src/services/productService.js
import api from "../api";

export const getHomeProducts = async () => {
    const response = await api.get("/products/active");
    return response.data;
};

export const getProductDetail = async (id) => {
    const response = await api.get(`/products/detail/${id}`);
    return response.data;
};
export const addToCart = async (userId, sellerProductId, quantity) => {
    const response = await api.post(`/cart/add?userId=${userId}`, {
        sellerProductId,
        quantity
    });
    return response.data;
};

File: modules/product/src/main/java/com/ecommerce/product/service/ProductService.java
package com.ecommerce.product.service;

import com.ecommerce.product.domain.Product;
import com.ecommerce.product.domain.SellerProduct;
import com.ecommerce.product.dto.request.ProductFilterRequest;
import com.ecommerce.product.dto.request.ProductRequest;
import com.ecommerce.product.dto.response.ProductDetailResponse;
import com.ecommerce.product.dto.response.ProductHomeResponse;
import org.springframework.data.domain.Page;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

public interface ProductService {
    Product createProduct(ProductRequest request);

    List<Product> getAll();

    Product getById(Long id);

    @Transactional
    Product updateProduct(Long id, ProductRequest request);

    @Transactional
    void deleteProduct(Long id);

    List<ProductHomeResponse> getAllActiveForHomePage(Long userId);

    List<Product> searchSemantic(List<Double> queryVector);

    SellerProduct getSellerProductById(Long id);

    void changeSellerProductStatus(Long sellerProductId, boolean active);


    List<SellerProduct> getAllActiveBySeller(Long sellerId);

    List<SellerProduct> getAllForAdmin();

    ProductDetailResponse getProductDetail(Long id);

    Page<ProductHomeResponse> filterProducts(
            ProductFilterRequest request,
            Long userId);
}